pkg/query/args.go

```diff
@@ -17,16 +18,26 @@ import (
 
 // NewArgs creates new query arguments with the defaults set
 func NewArgs(query, ifaces string, opts ...Option) *Args {
-	a := &Args{
-		// required args
-		Query:  query,
-		Ifaces: ifaces,
+	a := DefaultArgs()
 
-		// defaults
+	// required args
+	a.Query, a.Ifaces = query, ifaces
+
+	// apply functional options
+	for _, opt := range opts {
+		opt(a)
+	}
+	return a
+}
+
+// DefaultArgs creates a basic set of query arguments with only the
+// defaults being set
+func DefaultArgs() *Args {
+	return &Args{
 		First:      time.Now().AddDate(0, -1, 0).Format(time.ANSIC),
 		Format:     DefaultFormat,
 		In:         DefaultIn,
-		Last:       time.Now().Format(time.ANSIC),
+		Last:       time.Now().AddDate(0, 2, 0).Format(time.ANSIC),
```

Comment:
Why?

Reply:
Yeah, I hate this, too. Problem is: I need to ensure lateron that I reject live mode if an end timestamp has been set (assuming that it is before 'now()', precluding the use of live data). Thing is: I can't compare to `time.Now()` because by the time I hit that check, now has already passed. So I need something far enough in the future that I can check if _anything else_ is set. I first thought about setting the default to a "maximum timestamp" (which I can compare to), but turns out it's not so easy to actually determine what the maximum number for `time.Time` is (due to timezones and stuff there is not the one integer you can universally use). We could of course use some arbitrary timestamp in the far future...

Reply:
OK, took another look and now used a proper maximum timestamp defined in `types`. That one should be big / late enough and can be used for explicit comparison.